upup/pkg/fi/nodeup/command.go

```diff
@@ -363,6 +361,42 @@ func (c *NodeUpCommand) Run(out io.Writer) error {
 		klog.Exitf("error closing target: %v", err)
 	}
 
+	if api.CloudProviderID(c.cluster.Spec.CloudProvider) == api.CloudProviderAWS {
+		err := completeWarmingLifecycleAction(cloud.(awsup.AWSCloud), modelContext)
+		if err != nil {
+			return fmt.Errorf("failed to complete lifecylce action: %w", err)
+		}
+	}
+	return nil
+}
+
+func completeWarmingLifecycleAction(cloud awsup.AWSCloud, modelContext *model.NodeupModelContext) error {
+	asgName := modelContext.InstanceGroup.GetName() + "." + modelContext.Cluster.GetName()
+	hookName := "kops-warming"
+	svc := cloud.(awsup.AWSCloud).Autoscaling()
+	hooks, err := svc.DescribeLifecycleHooks(&autoscaling.DescribeLifecycleHooksInput{
+		AutoScalingGroupName: &asgName,
+		LifecycleHookNames:   []*string{&hookName},
+	})
+	if err != nil {
+		return err
+	}
+
+	klog.Infof("Checking if we should complete a lifecycle hook for %q", asgName)
```

Comment:
Maybe remove from here and add some message on the else branch below.
As is here, it says "Checking" and then nothing.